### DISCUSSION TITLE
Math optimization

### DISCUSSION THREAD

**Discussion by GillesLACAUD** [marcel-licence/esp32_basic_synth#67] [Q&A]
Hi Marcel
Could you please explain this line
//[[gnu::noinline, gnu::optimize ("fast-math")]]

Thank a lot

**Comment by marcel-licence** [comment]
I did some experiments to increase the performance. I think it was left from doing optimizations on the ESP8266
ref: https://www.esp32.com/viewtopic.php?t=800&start=20
I think this has no effect

**Reply by GillesLACAUD** [reply]
Thank you Marcel....
I have try this optimization : "build_type = release" with visual studio + platform IO and i have win something.
I do not know you to select the release build in the Arduino IDE.

**Reply by marcel-licence** [reply]
Good question. Do you have verbose output of the compiler statements? I can imagine that at the end both IDE's will do the same. 
I think there should be some output like
`...\\xtensa-esp32-elf-gcc\\1.22.0-97-gc752ad5-5.2.0/bin/xtensa-esp32-elf-g++"
-DESP_PLATFORM "-DMBEDTLS_CONFIG_FILE=\"mbedtls/esp_config.h\""
-DHAVE_CONFIG_H
-DGCC_NOT_5_2_0=0`
